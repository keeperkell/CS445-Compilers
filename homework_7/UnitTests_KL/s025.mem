Func: input returns type int [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 1  Func: output returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type int [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 2  Func: inputb returns type bool [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 3  Func: outputb returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type bool [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 4  Func: inputc returns type char [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 5  Func: outputc returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type char [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 6  Func: outnl returns type void [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 7  Var: runningSum of type int [mem: Global loc: 0 size: 1] [line: 3]
Sibling: 8  Func: sum returns type int [mem: Global loc: 0 size: -3] [line: 4]
.   Child: 0  Parm: x of type int [mem: Parameter loc: -2 size: 1] [line: 4]
.   Child: 1  Compound [mem: None loc: 0 size: -3] [line: 4]
.   .   Child: 0  Var: runningSum of static type int [mem: LocalStatic loc: -1 size: 1] [line: 5]
.   .   .   Child: 0  Const 0 of type int [line: 5]
.   .   Child: 1  Assign: += of type int [line: 7]
.   .   .   Child: 0  Id: runningSum of static type int [mem: LocalStatic loc: -1 size: 1] [line: 7]
.   .   .   Child: 1  Id: x of type int [mem: Parameter loc: -2 size: 1] [line: 7]
.   .   Sibling: 1  Return [line: 8]
.   .   .   Child: 0  Id: runningSum of static type int [mem: LocalStatic loc: -1 size: 1] [line: 8]
Sibling: 9  Func: main returns type void [mem: Global loc: 0 size: -2] [line: 11]
.   Child: 1  Compound [mem: None loc: 0 size: -2] [line: 12]
.   .   Child: 1  Assign: <- of type int [line: 13]
.   .   .   Child: 0  Id: runningSum of type int [mem: Global loc: 0 size: 1] [line: 13]
.   .   .   Child: 1  Const 7000 of type int [line: 13]
.   .   Sibling: 1  Call: output of type void [line: 14]
.   .   .   Child: 0  Id: runningSum of type int [mem: Global loc: 0 size: 1] [line: 14]
.   .   Sibling: 2  Call: sum of type int [line: 16]
.   .   .   Child: 0  Const 666 of type int [line: 16]
.   .   Sibling: 3  Call: sum of type int [line: 17]
.   .   .   Child: 0  Const 3 of type int [line: 17]
.   .   Sibling: 4  Call: output of type void [line: 18]
.   .   .   Child: 0  Call: sum of type int [line: 18]
.   .   .   .   Child: 0  Const 0 of type int [line: 18]
.   .   Sibling: 5  Call: output of type void [line: 20]
.   .   .   Child: 0  Id: runningSum of type int [mem: Global loc: 0 size: 1] [line: 20]
Offset for end of global space: -2
Number of warnings: 0
Number of errors: 0
