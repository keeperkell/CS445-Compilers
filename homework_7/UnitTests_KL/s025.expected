Number of warnings: 0
Number of errors: 0
------------------------------------
Loading file: UnitTests/s025.tm
7000 669 7000
Bye.
